### PR TITLE
[lang] Rescale_indix api and test

### PR DESCRIPTION
**Pull request by KLozes**
Related issue = #2177 
A simple index rescaling API that could be useful once we switch to consecutive indices.  #2248 

[[Click here for the format server]](http://kun.csail.mit.edu:31415/)

----

### CODE REVIEW

File: python/taichi/lang/ops.py [hunk 1]
```diff
@@ -546,3 +547,24 @@ def deactivate(l, indices):
...
+            f"first arguement must be a field"
+    assert isinstance(b, Expr) and b.is_global(), \
+            f"second arguement must be a field"
+    assert isinstance(I, matrix.Matrix) and not I.is_global(), \
+            f"third arguement must be a grouped index"
+    assert len(a.shape) >= len(b.shape)
+    Ib = I.copy()
+    for n in range(len(b.shape)):
+        if a.shape[n] > b.shape[n]:
+            assert (a.shape[n] // b.shape[n]) % 2 == 0, \
```
Comment (k-ye) [inline python/taichi/lang/ops.py:563]: Do we have to enforce `% 2 == 0`?

File: python/taichi/lang/ops.py [hunk 2]
```diff
@@ -546,3 +547,24 @@ def deactivate(l, indices):
 
 def length(l, indices):
     return Expr(_ti_core.insert_len(l.snode.ptr, make_expr_group(indices)))
+
+
+def rescale_index(a, b, I):
```
Comment (k-ye) [inline python/taichi/lang/ops.py:552]: nit: Could you add a docstring like https://github.com/taichi-dev/taichi/blob/abd063c92fa24616118c9feddcc13efc4b1c75fb/python/taichi/lang/__init__.py#L293-L312? Thanks!

**Review by k-ye — APPROVED** [review]
LGTM, thanks!

---

It seems that the first commit is not properly associated with your github account? Could you fix that issue so that CLA bot stops complaining?

File: python/taichi/lang/ops.py (hunk 1 above)
Comment (KLozes) [reply python/taichi/lang/ops.py:563]: Good question! Probably no.

File: python/taichi/lang/ops.py (hunk 2 above)
Comment (KLozes) [reply python/taichi/lang/ops.py:552]: Sure thing.

**Comment by KLozes** [comment]
> It seems that the first commit is not properly associated with your github account? Could you fix that issue so that CLA bot stops complaining?

Sorry about this! I'm not really sure there's a good way to fix this though. I could just create a whole new PR if you would like?

**Pull request closed without merging**